### PR TITLE
OD-558 [Fix] Time/Date picker validation

### PR DESCRIPTION
**Pull request by skrupnyk**
@sofiiakvasnevska @inna-bieshulia

OD-558 https://weboo.atlassian.net/browse/OD-558

## Description
**Problem:** Validation for date picker and time picker works in settings mode.
**Solution:** Changed value handling so that validation works only in preview mode.

## Screenshots/screencasts
https://storyxpress.co/video/kt47xikfs47k5fnfi

## Backward compatibility
This change is fully backward compatible.

Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/components/date.js [hunk 1]
```diff
@@ -95,12 +93,15 @@ Fliplet.FormBuilder.field('date', {
...
+      if (this.datepicker) {
+        this.datePicker.datepicker('setDate', val);
+      }
+
+      if (this.$v.value.$invalid) {
+        this.highlightError();
       }
 
-      this.highlightError();
-      this.$emit('_input', this.name, val, false, true);
```
Comment (tonytlwu) [inline js/components/date.js:103]: Why are the `false, true` parameters removed? They were added as part of https://github.com/Fliplet/fliplet-widget-form-builder/pull/409

**Review by tonytlwu — APPROVED** [review]

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Needs clarification

File: js/components/date.js (hunk 1 above)
Comment (upplabs-alex-levchenko) [reply js/components/date.js:103]: We've removed 'updateValue' usages which potentially were calling a loop of 'emit' events and now we are firing one 'emit' after the component load and one each time after value is updated.
Should we ask QA to check the issue for you previous fix? If yes, please let me know which one :)

Comment (tonytlwu) [reply js/components/date.js:103]: The bug was https://weboo.atlassian.net/browse/ID-1394 Please make sure QA verifies the bug is not re-introduced.

**Comment by skrupnyk** [comment]
new PR https://github.com/Fliplet/fliplet-widget-form-builder/pull/412

**Pull request closed without merging**